src/Raven.Server/Web/System/SetupHandler.cs

```diff
@@ -512,56 +523,33 @@ public async Task SetupUnsecured()
                     // ignored
                 }
 
-                var setupInfo = JsonDeserializationServer.UnsecuredSetupInfo(setupInfoJson);
+                var unsecuredSetupInfo = JsonDeserializationServer.UnsecuredSetupInfo(setupInfoJson);
 
-                BlittableJsonReaderObject settingsJson;
-                await using (var fs = new FileStream(ServerStore.Configuration.ConfigPath, FileMode.Open, FileAccess.Read))
-                {
-                    settingsJson = await context.ReadForMemoryAsync(fs, "settings-json");
-                }
-
-                settingsJson.Modifications = new DynamicJsonValue(settingsJson)
+                var operationResult = await ServerStore.Operations.AddOperation(
+                    null,
+                    "Setting up RavenDB in unsecured mode.",
+                    Documents.Operations.Operations.OperationType.Setup,
+                    progress => SetupManager.SetupUnsecuredTask(progress,
+                        unsecuredSetupInfo,
+                        ServerStore,
+                        context,
+                        operationCancelToken.Token),
+                    operationId.Value,
+                    token: operationCancelToken);
+
+                if (unsecuredSetupInfo.LocalNodeTag != null)
                 {
-                    [RavenConfiguration.GetKey(x => x.Licensing.EulaAccepted)] = true,
-                    [RavenConfiguration.GetKey(x => x.Core.SetupMode)] = nameof(SetupMode.Unsecured),
-                    [RavenConfiguration.GetKey(x => x.Security.UnsecuredAccessAllowed)] = nameof(UnsecuredAccessAddressRange.PublicNetwork)
-                };
-
-                if (setupInfo.Port == Constants.Network.ZeroValue)
-                    setupInfo.Port = Constants.Network.DefaultUnsecuredRavenDbHttpPort;
+                    var zip = ((SetupProgressAndResult)operationResult).SettingsZipFile;
 
-                settingsJson.Modifications[RavenConfiguration.GetKey(x => x.Core.ServerUrls)] = string.Join(";", setupInfo.Addresses.Select(ip => IpAddressToUrl(ip, setupInfo.Port)));
+                    var fileName = $"Unsecure.Cluster.Settings {DateTime.UtcNow:yyyy-MM-dd HH-mm}.zip ";
```

Comment:
wondering if we have anything better that we could put in the filename?

Reply:
What are you suggesting?